lib/shopify-cli/tasks/update_whitelist_url.rb

```diff
@@ -0,0 +1,39 @@
+module ShopifyCli
+  module Tasks
+    class UpdateWhitelistURL < ShopifyCli::Task
+      def call(ctx, url: ngrokurl)
+        @ctx = ctx
+        project = ShopifyCli::Project.current
+        app_type = project.app_type
+        api_key = Project.current.env.api_key
+        result = Helpers::PartnersAPI.query(ctx, 'get_app_urls', apiKey: api_key)
+        @callback = app_type.callback_url
+        app = result['data']['app']
+        whitelist_urls = check_urls(app['redirectUrlWhitelist'], url)
+        with_callback = check_callback_url(whitelist_urls, url)
+        return if with_callback == app['redirectUrlWhitelist']
+        ShopifyCli::Helpers::PartnersAPI.query(@ctx, 'update_whitelisturls', input: {
+          redirectUrlWhitelist: whitelist_urls, apiKey: api_key
+        })
+      end
+
+      def check_urls(urls, new_url)
+        urls.map do |url|
+          if (match = url.match(/https:\/\/([a-z0-9\-]+\.ngrok\.io)(.*)/))
+            "#{new_url}#{match[2]}"
+          else
+            url
+          end
+        end
+      end
+
+      def check_callback_url(urls, new_url)
+        if urls.grep(/callback/).empty?
```

Comment:
can we use the value from the app_type to search, rather than this hardcoded value?

Reply:
yes duh yes